diff --git include/linux/mm_types.h include/linux/mm_types.h
index 6c1dc6c..8c00faf 100644
--- include/linux/mm_types.h
+++ include/linux/mm_types.h
@@ -24,7 +24,7 @@
 
 struct address_space;
 
-#define USE_SPLIT_PTE_PTLOCKS	(NR_CPUS >= CONFIG_SPLIT_PTLOCK_CPUS)
+#define USE_SPLIT_PTE_PTLOCKS	(0)
 #define USE_SPLIT_PMD_PTLOCKS	(USE_SPLIT_PTE_PTLOCKS && \
 		IS_ENABLED(CONFIG_ARCH_ENABLE_SPLIT_PMD_PTLOCK))
 #define ALLOC_SPLIT_PTLOCKS	(SPINLOCK_SIZE > BITS_PER_LONG/8)
